examples/go/go-multi-rpc/integration/integration_test.go

```diff
@@ -0,0 +1,201 @@
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/magefile/mage/sh"
+	"github.com/tidwall/gjson"
+)
+
+func TestDeployed(t *testing.T) {
+	componentOneURN := mustGetCompURNByCompName(t, "component-one")
+	fmt.Printf("component-one: %s\n", componentOneURN)
+
+	componentTwoURN := mustGetCompURNByCompName(t, "component-two")
+	fmt.Printf("component-two: %s\n", componentTwoURN)
+
+	componentThreeURN := mustGetCompURNByCompName(t, "component-three")
+	fmt.Printf("component-three: %s\n", componentThreeURN)
+}
+
+func TestCallingAddOnComponentOneCallsToOtherComponents(t *testing.T) {
+	workerName := uuid.New().String()
+	fmt.Printf("random worker name for test: %s\n", workerName)
+
+	// Setup
+	{
+		// Get all component URNs
+		componentURNs := mustGetComponentURNs(t)
+		fmt.Printf("component urns: %#v\n", componentURNs)
+
+		// Preparing workers with env vars for RPC, so they know the other component IDs
+		mustAddComponent(t, "component-one", workerName, componentURNs)
+		mustAddComponent(t, "component-two", workerName, componentURNs)
+	}
+
+	// Call get on all component and check the counters are 0
+	{
+		expectCounter(t, "component-one", workerName, 0)
+		expectCounter(t, "component-two", workerName, 0)
+		expectCounter(t, "component-three", workerName, 0)
+	}
+
+	// Invoke add on component-one
+	{
+		mustInvokeAndAwaitComponent(t, "component-one", workerName, "pack-ns:component-one/component-one-api.{add}", []string{"3"})
+	}
+
+	// Call get on all component and check the counters are accumulated on component two and three
+	{
+		expectCounter(t, "component-one", workerName, 3)
+		expectCounter(t, "component-two", workerName, 3)
+		expectCounter(t, "component-three", workerName, 6)
+	}
+
+	// Invoke add on component-two
+	{
+		mustInvokeAndAwaitComponent(t, "component-two", workerName, "pack-ns:component-two/component-two-api.{add}", []string{"2"})
+	}
+
+	// Call get on all component and check the counters are accumulated on component three
+	{
+		expectCounter(t, "component-one", workerName, 3)
+		expectCounter(t, "component-two", workerName, 5)
+		expectCounter(t, "component-three", workerName, 8)
+	}
+
+	// Invoke add on component-one again
+	{
+		mustInvokeAndAwaitComponent(t, "component-one", workerName, "pack-ns:component-one/component-one-api.{add}", []string{"1"})
+	}
+
+	// Call get on all component and check the counters are accumulated on component two and three
+	{
+		expectCounter(t, "component-one", workerName, 4)
+		expectCounter(t, "component-two", workerName, 6)
+		expectCounter(t, "component-three", workerName, 10)
+	}
+}
+
+func getCompURNByCompName(compName string) (string, error) {
```

Comment:
this is a great touch. need something like this in other rpc examples (for other languages